docs/proposals/registry/index-server-REST-API.md

```diff
@@ -0,0 +1,243 @@
+# Index server REST APIs
+This documentation explains how to use the index server REST APIs
+
+## Gets registry index file
+Gets the registry index file content from HTTP response
+### HTTP request
+```
+GET http://{registry host}/index
+```
+### Request parameters
+| Parameter | Description |
+| --------- | ----------- |
+| registry host | the URL/ingress that exposes registry service |
+### Request body
+The request body must be empty.
+### Request example
+```
+curl http://devfile-registry.192.168.1.1.nip.io/index
+```
+### Response example
+```json
+[
+  {
+    "name": "java-maven",
+    "displayName": "Maven Java",
+    "description": "Upstream Maven and OpenJDK 11",
+    "tags": [
+      "Java",
+      "Maven"
+    ],
+    "projectType": "maven",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-maven:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
```

Comment:
Is it a really good example? I thought devfile and meta yamls are already merged into within https://github.com/devfile/api/pull/248
What are other possible resources apart from devfile.yaml?